graphql/query_resolver.go

```diff
@@ -1033,8 +1033,18 @@ func (r *queryResolver) Waterfall(ctx context.Context, options WaterfallOptions)
 		// If no order options were specified, we're on the first page and should not put a limit on the first version returned so that we don't omit inactive versions
 		maxVersionOrder = 0
 	} else if maxOrderOpt == 0 {
-		// If we're paginating backwards, use the newest active version as the upper bound
-		maxVersionOrder = activeVersions[0].RevisionOrderNumber
+		// Find the next active version. If it doesn't exist, we're on the first page. If it does,
+		// set the max order to one less than its order so that we fetch all of the inactive versions
+		// in between.
```

Comment:
Updated the comment and order of if-statements

I think the reason to subtract 1 is because I'm getting the next active version order, it means that when you subtract 1 it's either:
* The 0th version I already have in `activeVersions`
* The most recent inactive version within its collapsed group

Tested with 3 inactive commits, it seems to be working: 

https://github.com/user-attachments/assets/b7d397c7-1ffb-4f0e-a554-c9bc288d608b

Reply:
Ah yes gotcha, I think the backwards comments confused me. Thank you!